### ISSUE TITLE
"Link all assemblies" iOS option causes failure to load OData adapter

### ISSUE THREAD

**Issue by object** [object/Simple.OData.Client#106]
Described in the comment to the post:

https://vagifabilov.wordpress.com/2014/11/03/resolving-filenotfoundexception-when-using-simple-odata-client-in-ios-applications/comment-page-1/#comment-102

"However, I have trouble using it with third linker options: Link all assemblies.
Maybe the linker stripped some members from some other dependent assembly (e.g. Micrsoft.Data.OData)"

IUnable to load OData adapter from assembly Simple.OData.Client.V3.Adapter —> System.ArgumentNullException: Argument cannot be null. Parameter name: type at System.Reflection.IntrospectionExtensions.GetTypeInfo (System.Type type) [0x0000c] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Reflection/IntrospectionExtensions.cs:38 at Simple.OData.Client.Extensions.TypeExtensions.GetDeclaredConstructors (System.Type type) [0x00000] in :0 at Simple.OData.Client.AdapterFactory.LoadAdapter (System.String adapterAssemblyName, System.String adapterTypeName, System.Object[] ctorParams) [0x00000] in :0 — End of inner exception stack trace — at Simple.OData.Client.AdapterFactory.LoadAdapter (System.String adapterAssemblyName, System.String adapterTypeName, System.Object[] ctorParams) [0x00000] in :0 at Simple.OData.Client.AdapterFactory+c__DisplayClass18.b__16 () [0x00000] in :0 at Simple.OData.Client.AdapterFactory+d__0.MoveNext () [0x00000] in :0 — End of stack trace from previous location where exception was thrown — at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000b] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.ExceptionServices/ExceptionDispatchInfo.cs:62 at System.Runtime.CompilerServices.TaskAwaiter`1[Simple.OData.Client.IODataAdapter].GetResult () [0x00034] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.CompilerServices/TaskAwaiter_T.cs:59 at Simple.OData.Client.Session+d__c.MoveNext () [0x00000] in :0 — End of stack trace from previous location where exception was thrown — at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000b] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.ExceptionServices/ExceptionDispatchInfo.cs:62 at System.Runtime.CompilerServices.TaskAwaiter`1[Simple.OData.Client.IODataAdapter].GetResult () [0x00034] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Runtime.CompilerServices/TaskAwaiter_T.cs:59 at Simple.OData.Client.ODataClient+d__b6.MoveNext () [0x00000] in :0 } System.InvalidOperationException

Here is the stack trace:
at System.Threading.Tasks.Task`1[System.Collections.Generic.IEnumerable`1[System.Collections.Generic.IDictionary`2[System.String,System.Object]]].get_Result () [0x0003c] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Threading.Tasks/Task_T.cs:52
at Simple.OData.Client.FluentClientBase`1+c__DisplayClass4[Ptix.Entities.Show].b__3 (System.Threading.Tasks.Task`1 x) [0x00000] in :0
at System.Threading.Tasks.TaskActionInvoker+FuncTaskInvoke`2[System.Collections.Generic.IEnumerable`1[System.Collections.Generic.IDictionary`2[System.String,System.Object]],System.Collections.Generic.IEnumerable`1[System.Collections.Generic.IDictionary`2[System.String,System.Object]]].Invoke (System.Threading.Tasks.Task owner, System.Object state, System.Threading.Tasks.Task context) [0x00000] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Threading.Tasks/TaskActionInvoker.cs:330
at System.Threading.Tasks.Task.InnerInvoke () [0x00019] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Threading.Tasks/Task.cs:546
at System.Threading.Tasks.Task.ThreadStart () [0x000d0] in /Developer/MonoTouch/Source/mono/mcs/class/corlib/System.Threading.Tasks/Task.cs:438

**Comment by Mabux** [comment]
Hi, any progress on this issue? 
I now get the exact identical StackTrace in a Xamarin Class-Library, independent of Device.
It is caused when calling 
`var commandText = await command.GetCommandTextAsync();`
As we have checked in all packages on 15.April (Simple.OData.V3 in version 4.3.0), and have not updated them since, there no changes on this, or its dependencies, could be responsible for the exception.
Only updates we did were Xamarin/SDK, and .NET/Windows.
